rllib/models/configs/encoder.py

```diff
@@ -0,0 +1,83 @@
+import abc
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Tuple
+
+from ray.rllib.models.specs.specs_dict import ModelSpec
+from ray.rllib.models.torch.encoders.vector import TorchVectorEncoder
+
+if TYPE_CHECKING:
+    from ray.rllib.models.torch.encoders.vector import Encoder
```

Comment:
If importing `from ray.rllib.models.torch.encoders.vector import TorchVectorEncoder` works, we should also move `Encoder` out of this TYPE_CHECKING if-block. Then we won't need the "" quotes anymore either.

Reply:
We actually need this to prevent circular imports when typing. `VectorEncoderConfig.build()` will return a `TorchVectorEncoder` and `TorchVectorEncoder` takes a `VectorEncoderConfig` as input.

Reply:
@sven1977 maybe this is a reason to define the implementation in the same file as the config?